src/Shape.cpp

```diff
@@ -23,6 +24,80 @@ bool Shape::intersect(const Ray &, float t0, float t1, double& distance) const {
 }
 
 
+/** @brief Shape::contains() determines whether the shape intersects a sphere with center p.
```

Comment:
IMHO we should add these docstrings to the header rather than the implementation.

Nice that you added the reference!

Reply:
Done